File: scripts/modules/lua/common/FileLinesIterator_1_0.lua
FileLinesIterator = {}

function FileLinesIterator:new(errors)
	newObj = {
		errors = errors,
		file = RWFile:new(errors),
		eof = false,
		line = nil
	}
	self.__index = self
	return setmetatable(newObj, self)
end

function FileLinesIterator:read_line()
	self.line = self.file:read_line()
	if self.line == nil then
		self.eof = true
	end 
end

function FileLinesIterator:open_file(file_path)
	self.file:open_file(file_path, 'r')
	if self.errors.error_occured then
		self.eof = true
	else
		self:read_line()
	end
end

function FileLinesIterator:close_file()
	self.file:close_file()
end

function FileLinesIterator:next_line()
	local line = self.line
	self:read_line()
	return line
end

File: scripts/modules/python/common/FileLinesIterator_1_0.py
# -*- coding: utf-8 -*

from common.RWFile_1_0 import *

class FileLinesIterator:

	def __init__(self, errors):
		self.errors = errors
		self.file = RWFile(errors)
		self.eof = False
		self.line = None

	def read_line(self):
		self.line = self.file.read_line()
		if self.line == None:
			self.eof = True

	def open_file(self, file_path):
		self.file.open_file(file_path, 'r')
		if self.errors.error_occured:
			self.eof = True
		else:
			self.read_line()

	def close_file(self):
		self.file.close_file()

	def next_line(self):
		line = self.line
		self.read_line()
		return line
